pluginframework/resource_quality_monitor.go

```diff
@@ -0,0 +1,238 @@
+	resp.Schema = schema.Schema{
+		Description: "Terraform schema for Databricks Lakehouse Monitor. MonitorInfo struct is used to create the schema",
+		Attributes: common.PluginFrameworkResourceStructToSchemaMap(MonitorInfoExtended{}, func(c common.CustomizableSchemaPluginFramework) common.CustomizableSchemaPluginFramework {
+			c.SetRequired("assets_dir")
+			c.SetRequired("output_schema_name")
+			c.SetRequired("table_name")
+			c.SetReadOnly("monitor_version")
+			c.SetReadOnly("drift_metrics_table_name")
+			c.SetReadOnly("profile_metrics_table_name")
+			c.SetReadOnly("status")
+			c.SetReadOnly("dashboard_id")
+			c.SetReadOnly("schedule", "pause_status")
+			return c
+		}),
+	}
+}
+
+func (d *QualityMonitorResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	client, ok := req.ProviderData.(*common.DatabricksClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *common.DatabricksClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	d.Client = client
+}
+
+func (r *QualityMonitorResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	w, err := r.Client.WorkspaceClient()
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get workspace client", err.Error())
+		return
+	}
+	var monitorInfoTfSDK MonitorInfoExtended
+	diags := req.Plan.Get(ctx, &monitorInfoTfSDK)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var createMonitorGoSDK catalog.CreateMonitor
+	err = common.TfSdkToGoSdkStruct(monitorInfoTfSDK, &createMonitorGoSDK, ctx)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to convert Tf SDK struct to Go SDK struct", err.Error())
+		return
+	}
+	endpoint, err := w.QualityMonitors.Create(ctx, createMonitorGoSDK)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get created monitor", err.Error())
+		return
+	}
+	err = WaitForMonitor(w, ctx, endpoint.TableName)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to wait for newly created monitor", err.Error())
+		return
+	}
+
+	// Get the created monitor.
+	new_endpoint, err := w.QualityMonitors.GetByTableName(ctx, createMonitorGoSDK.TableName)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get newly created monitor", err.Error())
+		return
+	}
+
+	var newMonitorInfoTfSDK MonitorInfoExtended
+	err = common.GoSdkToTfSdkStruct(new_endpoint, &newMonitorInfoTfSDK, ctx)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to convert Go SDK struct to TF SDK struct", err.Error())
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, newMonitorInfoTfSDK)...)
+}
+
+func (r *QualityMonitorResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	w, err := r.Client.WorkspaceClient()
```

Comment:
We can expose another method on DatabricksClient to return `diag.Diagnostics` to follow the same pattern, but it would probably look like:
```go
func (c *Client) GetWorkspaceClient(w **databricks.WorkspaceClient) diag.Diagnostics {
    *w = ...
    return nil
}

...

var w *databricks.WorkspaceClient
resp.Diagnostics.Append(r.Client.GetWorkspaceClient(&w))
if resp.Diagnostics.HasError() {
    return
}
```

or if this is too overkill, we can do
```go
func (c *Client) GetWorkspaceClient() (*databricks.WorkspaceClient, diag.Diagnostics) {...}
```